### PR TITLE
BTFSINFRA-533 add make test_go_mod option

### PR DESCRIPTION
**Pull request by richlee33**
add a new script to run "go mod tidy" and then check for diffs.  return exit code 1 (fail) if diffs are found and added to test_go_mod make option.  also added this to the make test_short option.

### CODE REVIEW

File: mk/golang.mk [hunk 1]
```diff
@@ -68,6 +68,10 @@ test_go_fmt:
 .PHONY: test_go_fmt
 TEST_GO += test_go_fmt
 
+test_go_mod: 
+	go mod tidy
```
Comment (taiyangc) [inline mk/golang.mk]: `test_go_mod` should check after running `go mod tidy`, whether `git diff` has any resulted changes, similar to what `test-go-fmt` does
https://github.com/TRON-US/go-btfs/blob/master/bin/test-go-fmt

This way devs wouldn't commit out-of-date deps

Comment (richlee33) [reply mk/golang.mk]: ah ok i understand.  thanks.  i will redo the PR.

**Comment by richlee33** [comment]
@taiyangc i think its ready to go now.  take a look at your convenience?

**Comment by taiyangc** [comment]
@richlee33 Code looks good to me. Could you add `test_go_fmt` and `test_go_mod` to jenkins job and make tests pass? thanks!

**Comment by richlee33** [comment]
@taiyangc 
i am now running in the docker container the following cmd:
"Step 21/21 : CMD make test_go_fmt test_go_mod test_go_test"
so there is only 1 build step in the jenkins job.  1 stop shop. 😃 
it ran fine:
https://builds.bittorrent.com/view/btfs/job/btfs-dev-test/62/console

**Pull request merged**